### DISCUSSION TITLE
HTML Mode is also filtering inverse, paragraphs, etc.

### DISCUSSION THREAD

**Discussion by omittelstaedt** [tomik23/show-more#74] [Ideas]
First of all, thanks for your great work. 

I would like to use this lib for an html block and it filters a lot of html tags out of the original text. For example:

`<p>This is a paragraph with <b>bold</b> and <i>italic</i>Text.</p>` 

becomes:

`This is a paragraph with <b>bold</b> and italic Text.` 

This has the effect, that my truncated text is not css formated any more, i.e. paragraphs have no padding afterwards.

To avoid this, I hacked 2 changes in the lib and now it works for me. First of all, I think it is better to use this solution for the function htmlString:

https://stackoverflow.com/questions/6003271/substring-text-with-html-tags-in-javascript#6004268

Could you use this html function in your lib or provide a callback function as option, so that I could use my own?

Then I turned off all regex replacements (I don't know why these are needed anyway, the htmlString function does a great job and in my opinion, nothing else needs to be done here. So could you add a way to deactivate the default regex replacements?

After these 2 changes, your lib works perfect for html (at least in my cases). 

Thanks again for your work!

**Comment by tomickigrzegorz** [comment]
Thanks for the suggestion. I'll see what I can do.
Have you tested this solution with stackoverflow with all the html elements available, e.g. <img, li, div, etc.?

I'll put it into the discussion because it's not a issue :)

PS.
Unfortunately, this function does not work correctly, I mean counting the number of characters. I checked how it handles plain text with a small number of HTML characters.
I checked the example where the text "Contrary to popular belief. Lorem ..."
I set the limit to 510 characters, my code works correctly, up to exactly 510 characters, and the one you provided is not true, and it's very bad because it cuts to 438 characters.
I know it's better because it has formatting, but when it comes to character counting, unfortunately it doesn't work properly. I don't have time to analyze this code yet, but maybe I can turn off the formatting that is in the library and use my own function to remove characters.

**Reply by omittelstaedt** [reply]
The character counting was not that important for me so I haven't checked for the correct number. But the formatting was a must-have for my customer, so I made these changes. It works with html `<p>, <ul>, <li>, <strong>, <b>, <span>`. I haven't tested it with other tags.

**Reply by tomickigrzegorz** [reply]
Thanks for the information. I will try to disable the default regex, and enable the calback function to modify the text.
But as I wrote above, when I find time ;)
